### ISSUE TITLE
Schema parse error with `rebar3 release`

### ISSUE THREAD

**Issue by gdhgdhgdh** [erlio/vernemq#263]
### Environment

- VerneMQ Version: 0.15.3
- OS:Docker container
- Erlang/OTP version (if building from source): the public `erlang:19` docker image

### Expected behavior

As an Erlang newbie I am experimenting with adding extra application to test releases + upgrades without downtime. I have a few days of experience with VerneMQ so am using it as a base for this work.

As part of this, I would expect that `rebar3 release` completes successfully, but it does not, even when working on `git checkout 0.15.3` of the vernemq repo.

### Actual behaviour

All of the commands were run inside a vernemq 0.15.3 live container, which was built using the public `erlang:19` container.

```
root@51d4af2dd37b:/host# git status
HEAD detached at 0.15.3
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   rebar.lock

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	rebar3.crashdump

no changes added to commit (use "git add" and/or "git commit -a")
root@51d4af2dd37b:/host# rebar3 compile
===> Verifying dependencies...
===> Compiling vmq_plugin
===> Compiling vmq_commons
===> Compiling vmq_server
===> Compiling vmq_acl
===> Compiling vmq_passwd
make: Entering directory '/host/apps/vmq_passwd/c_src'
gcc  vmq_passwd.c -lcrypto -o ../priv/vmq_passwd
make: Leaving directory '/host/apps/vmq_passwd/c_src'
===> Compiling vmq_bridge
root@51d4af2dd37b:/host# rebar3 release
===> Verifying dependencies...
===> Compiling vmq_plugin
===> Compiling vmq_commons
===> Compiling vmq_server
===> Compiling vmq_acl
===> Compiling vmq_passwd
make: Entering directory '/host/apps/vmq_passwd/c_src'
gcc  vmq_passwd.c -lcrypto -o ../priv/vmq_passwd
make: Leaving directory '/host/apps/vmq_passwd/c_src'
===> Compiling vmq_bridge
===> Running cuttlefish schema generator
===> Starting relx build process ...
===> Resolving OTP Applications from directories:
          /host/_build/default/lib
          /host/apps
          /usr/local/lib/erlang/lib
          /host/_build/default/rel
===> Resolved vernemq-0.15.3
===> Including Erts from /usr/local/lib/erlang
===> release successfully created!
Schema parse error near line number 298: syntax error before: '}'
Schema parse error near line number 406: syntax error before: '}'
Schema parse error near line number 516: syntax error before: '}'
Schema parse error near line number 522: syntax error before: '}'
Schema parse error near line number 528: syntax error before: '}'
Schema parse error near line number 534: syntax error before: '}'
Schema parse error near line number 586: syntax error before: '}'
Schema parse error near line number 591: syntax error before: '}'
Schema parse error near line number 602: syntax error before: '}'
Schema parse error near line number 620: syntax error before: '}'
Error parsing schema: /host/_build/default/rel/vernemq/share/schema/vmq_server.schema
===> Uncaught error in rebar_core. Run with DEBUG=1 to see stacktrace or consult rebar3.crashdump
===> When submitting a bug report, please include the output of `rebar3 report "your command"`
root@51d4af2dd37b:/host# cat rebar3.crashdump 
Error: function_clause
[{cuttlefish_schema,'-merger/1-fun-0-',
     [{#Fun<cuttlefish_schema.0.55980616>,
       "/host/_build/default/rel/vernemq/share/schema/vmq_plugin.schema"},
      {errorlist,
          [{error,{erl_parse,{"syntax error before: '}'",298}}},
           {error,{erl_parse,{"syntax error before: '}'",406}}},
           {error,{erl_parse,{"syntax error before: '}'",516}}},
           {error,{erl_parse,{"syntax error before: '}'",522}}},
           {error,{erl_parse,{"syntax error before: '}'",528}}},
           {error,{erl_parse,{"syntax error before: '}'",534}}},
           {error,{erl_parse,{"syntax error before: '}'",586}}},
           {error,{erl_parse,{"syntax error before: '}'",591}}},
           {error,{erl_parse,{"syntax error before: '}'",602}}},
           {error,{erl_parse,{"syntax error before: '}'",620}}}]}],
     [{file,
          "/host/_build/default/plugins/cuttlefish/src/cuttlefish_schema.erl"},
      {line,60}]},
 {lists,foldr,3,[{file,"lists.erl"},{line,1276}]},
 {lists,foldr,3,[{file,"lists.erl"},{line,1276}]},
 {cuttlefish_schema,merger,1,
     [{file,
          "/host/_build/default/plugins/cuttlefish/src/cuttlefish_schema.erl"},
      {line,59}]},
 {rebar3_cuttlefish_release,do,1,
     [{file,
          "/host/_build/default/plugins/rebar3_cuttlefish/src/rebar3_cuttlefish_release.erl"},
      {line,95}]},
 {rebar_core,do,2,
     [{file,"/usr/src/rebar3-src/_build/default/lib/rebar/src/rebar_core.erl"},
      {line,125}]},
 {rebar3,main,1,
     [{file,"/usr/src/rebar3-src/_build/default/lib/rebar/src/rebar3.erl"},
      {line,56}]},
 {escript,run,2,[{file,"escript.erl"},{line,757}]}]
```

Please be gentle, this is all devastatingly new to me....

**Comment by ioolkos** [comment]
Hi @gdhgdhgdh hm... the error is not clear to me immediately.
In any case you should use `make rel`to build VerneMQ.

(We're all constantly learning, don't worry about this Erlang newbie thing! :)

**Comment by dergraf** [comment]
@gdhgdhgdh are you using the rebar3 coming with VerneMQ or did you use an own rebar3 binary somewhere on your PATH? I feel I've seen this error before with a newer version of rebar3, but that's just a guess. ;)

**Comment by gdhgdhgdh** [comment]
Ah @dergraf thank you. I can see that `make rel` is adding extra parameters to the `rebar3 release` command:

```root@e6383587c655:/host# make rel        
/host/rebar3  release --overlay_vars vars.config
===> Package <<"rebar3_cuttlefish">> not found. Fetching registry updates and trying again...
===> Updating package registry...
===> Writing registry to /root/.cache/rebar3/hex/default/registry
===> Generating package index...
[....]
===> Starting relx build process ...
===> Resolving OTP Applications from directories:
          /host/_build/default/lib
          /host/apps
          /usr/local/lib/erlang/lib
===> Resolved vernemq-0.15.3
===> Including Erts from /usr/local/lib/erlang
===> release successfully created!
```

FWIW, it does also work using the current 3.3.x series of `rebar3` - it might be worth updating the old `3.0.0-beta` ? :)
